lib/teleterm/apiserver/handler/handler_access_requests.go

```diff
@@ -214,6 +215,10 @@ func newAPIAccessRequest(req clusters.AccessRequest) *api.AccessRequest {
 		ResourceIds:             requestedResourceIDs,
 		Resources:               resources,
 		PromotedAccessListTitle: req.GetPromotedAccessListTitle(),
+		AssumeStartTime:         getProtoTimestamp(req.GetAssumeStartTime()),
+		MaxDuration:             timestamppb.New(req.GetMaxDuration()),
+		RequestTtl:              timestamppb.New(req.Expiry()),
+		SessionTtl:              timestamppb.New(req.GetSessionTLL()),
```

Comment:
The changes from this PR seem okay and they typecheck. I'm able to successfully submit an access request, but when I then go to list all requests in Connect, I see "Invalid time value".

I added a `console.error` to see the exact error:

```diff
diff --git a/web/teleterm/src/ui/DocumentAccessRequests/useAccessRequests.tsx b/web/teleterm/src/ui/DocumentAccessRequests/useAccessRequests.tsx
index 216dae57..d03f6e36 100644
--- a/web/teleterm/src/ui/DocumentAccessRequests/useAccessRequests.tsx
+++ b/web/teleterm/src/ui/DocumentAccessRequests/useAccessRequests.tsx
@@ -57,6 +57,7 @@ export default function useAccessRequests(doc: types.DocumentAccessRequests) {
       );
       setAccessRequests(requests);
     } catch (err) {
+      console.log(err);
       setAttempt({
         status: 'failed',
         statusText: err.message,

```

And the error is:

```
RangeError: Invalid time value
    at formatDistanceStrict (date-fns.js?v=1abd8bfc:2759:11)
    at getDurationText (makeAccessRequest.ts:87:20)
    at makeAccessRequest (makeAccessRequest.ts:24:22)
    at makeUiAccessRequest (useAccessRequests.tsx:98:10)
    at useAccessRequests.tsx:56:17
    at Array.map (<anonymous>)
    at getRequests (useAccessRequests.tsx:55:33)
```

It looks like `makeAccessRequest` from `e/web/teleport/src/services/workflow/makeAccessRequest.ts` expects dates to be `Date` instances. The argument to it doesn't have a type, so we didn't get type errors for that.

Perhaps `makeUiAccessRequest` from `e/web/teleterm/src/ui/DocumentAccessRequests/useAccessRequests.tsx` needs to be updated?

Reply:
thanks so much for catching these, i updated the types [here](https://github.com/gravitational/teleport.e/pull/3641/commits/0d7959a817b93b12fbcdc6063aaf0c7af1173f88) and added a test to hopefully catch them easier in future. probably the best way to prevent is to type the response value right? but i think that's going to create more refactoring